### DISCUSSION TITLE
Always unhide `.github/`

### DISCUSSION THREAD

**Discussion by peterbe** [sharkdp/fd#1226] [Q&A]
```
❯ fd  test-translations
# note, nothing found

❯ fd --hidden test-translations
.github/workflows/test-translations.yml
```

But if I have to use `--hidden`, it'll also show a bunch of other "junk". 
How can I train `fd` to not treat `.github` (for example) as a hidden directory? Ideally in a global config file because I don't want to have to remember copy some local `.fdignore` file into every `git` repo root. 

PS. `.fdignore` might not at all be the right direction anyway because it's not that I want to ignore more, I want to ignore less kinda.

**Comment by tmccombs** [comment]
I think you could do:

    --hidden --excluse ".*" --exclude "!.github"

That is enable searching hidden files, but exclude all files that start with "." except for . github

**Reply by peterbe** [reply]
Doesn't work :(

```
❯ fd --hidden --exclude '.*' --exclude '!\.github' dependabot.yml
content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
data/reusables/dependabot/check-in-dependabot-yml.md
```
Contrast with
```
❯ fd --hidden dependabot.yml
content/code-security/dependabot/dependabot-version-updates/configuration-options-for-the-dependabot.yml-file.md
data/reusables/dependabot/check-in-dependabot-yml.md
.github/dependabot.yml
```

**Reply by tmccombs** [reply]
it would be `!.github` not `!\.github`. But that doesn't work for me either. I wonder if that is a bug in the ignore crate

**Reply by tmccombs** [reply]
oh, duh. `fd --hidden --exclude '.*' --exclude '!.github/**' dependabot.yml` works.  (you could also probably use `!.github*`

**Reply by peterbe** [reply]
Doesn't work either. Sadly. Thanks for trying to help, I really appreciate it. 

```
# Expected
❯ fd dependabot.yml

# Also expected
❯ fd --hidden dependabot.yml
.github/dependabot.yml

# Unfortunately
❯ fd --hidden --exclude '.*' --exclude '!.github/**' dependabot.yml

```

**Reply by tmccombs** [reply]
huh, I could have sworn I got it to work with that. but now it isn't working for me either. Although I don't know why. 

FWIW, 

```
rg --hidden --files -g '!.*' -g '.github/**'
``` 
also doesn't work, which makes me wonder if the ignore crate isn't properly handling adding back something that is ignored by another rule.

**Reply by tmccombs** [reply]
Interesting. So if I create a `.ignore` file that looks like:

```gitignore
.*
!.github
```
then run  `fd --hidden dependabot.yml` then it finds the file. 

However, if I run `fd --hidden --exclude '.*' --exclude '!.github' dependabot.yml` then it doesn't find it.

**Reply by peterbe** [reply]
> then run `fd --hidden dependabot.yml` then it finds the file.

Well, doesn't that just ignore (no pun intended) any filtering of the sorts. So it's like `sudo` mode.

**Reply by tmccombs** [reply]
No `--hidden` just enables searching hidden files and directories (files that start with "."). It doesn't turn off the parsing ignore files, you would need `--no-ignore` for that.

**Comment by airled** [comment]
Hi there. Seems like it is not possible at the moment. But you can use a not so readable non-fd way with grep. Something like this:
```bash
fd -H test | grep -E -v '(\.\/\.)(?<!github)'
```
You use `-H` here to match also hidden files and directories. Then you filter fd's output with grep which excludes all hidden entries except github using negative lookbehind assertion. Should help you for time.
